apache/incubator-zeppelin
# [ZEPPELIN-773] Adding support for spark interpreter to work with yarn…

### What is this PR for?
Adding support for zeppelin to run in yarn-cluster mode through Livy
- Creating new interpreter (livy-spark) with (spark, pyspark, sparkr)
- Extending the interpreter to allow manipulation in the configurations of Spark from zeppelin web ui.


### What type of PR is it?
Feature
### Todos
* [Test case ] - Task
* [Documentation ] - Task

### What is the Jira issue?
* [ZEPPELIN-773]


### How should this be tested?
- Install Livy on https://github.com/cloudera/hue/tree/master/apps/spark/java
- Start the Livy server
- Start the Zeppelin and go to interpreter page, create livy interpreter with supported configurations
- Run livy-interpreter on any of three modes (`%livy.spark` to spark, `%livy.pyspark` to python, `%livy.sparkr` to R)

***If you run on a cluster, the cluster should support yarn, and run livy-server in yarn mode***


### Screenshots (if appropriate)

![](http://s32.postimg.org/ofgw4nabp/Screenshot_from_2016_05_15_10_48_48.png)
![](http://s32.postimg.org/7janekd8l/Screenshot_from_2016_05_18_14_00_48.png)


### Questions:
* Does the licenses files need update? no
* Is there breaking changes for older versions? no
* Does this needs documentation? yes
